src/cascadia/TerminalCore/TerminalSelection.cpp

```diff
@@ -832,8 +839,6 @@ void Terminal::ClearSelection()
 // - wstring text from buffer. If extended to multiple lines, each line is separated by \r\n
 const TextBuffer::TextAndColor Terminal::RetrieveSelectedTextFromBuffer(bool singleLine)
 {
-    auto lock = LockForReading();

```

Comment:
note to self: watch to see where the read lock in `RetrieveSelectedText` goes

Reply:
Into `ControlCore` here: https://github.com/microsoft/terminal/blob/dev/lhecker/9617-terminal-race-conditions/src/cascadia/TerminalControl/ControlCore.cpp#L1168
And one more down below. This PR moves all locking outside of `Terminal`.